src/gpu/intel/ocl/rnn/rnn_grid.cl

```diff
@@ -1005,28 +1005,57 @@ void cell_common_inner(const_wei_layer_cell_t wei_layer,
         for (int c_l = 0; c_l < CELL_DHC_THR; c_l++) {
             int c = c_thr + c_l * c_thr_stride;
             if (CELL_DHC_TAIL && c >= dims.dhc) break;
-            if (CELL_KIND == VANILLA_LSTM) {
-                float G[vanilla_lstm_n_gates];
-                float B[vanilla_lstm_n_bias];
-                for (int gate_idx = 0; gate_idx < vanilla_lstm_n_gates;
-                        gate_idx++) {
-                    G[gate_idx] = C[gate_idx][n_l][c_l];
-                    B[gate_idx] = convert_float(
-                            ctx.lstm.bias[off_ker_bias(dims.dhc, gate_idx, c)]);
-                }
-                vanilla_lstm_gates_t g
-                        = vanilla_lstm_compute_gates(G, B, ctx.lstm.tm_scales);
-                vanilla_lstm_store(gates.ptr, gates.strides.mb, states.ptr,
-                        ctx.lstm.c_states, ctx.lstm.c_states_iter,
-                        states.strides.mb, dims.dhc, n, c, ctx.lstm.tm_cscale,
-                        g);
-            } else if (CELL_KIND == VANILLA_RNN) {
-                float g = vanilla_rnn_compute_gates(C[0][n_l][c_l],
-                        ctx.rnn.bias[off_ker_bias(dims.dhc, 0, c)],
-                        ctx.rnn.alpha, ctx.rnn.tm_scales);
-                store_vanilla_rnn(gates.ptr, gates.strides.mb, states.ptr,
-                        states.strides.mb, dims.dhc, n, c, g);
```

Comment:
Please keep the runtime `if` statements so that we get compiler tests on all cases during development.